### ISSUE TITLE
else-if-used behaviour is incorrect in instances involving f-strings

### ISSUE THREAD

**Issue by nth10sd** [PyCQA/pylint#2528]
### Steps to reproduce
1. Make this your `.pylintrc`:
```
[MASTER]
load-plugins=pylint.extensions.check_elif,
```
2. Create the following testcase:
```
"""Minimised testcase showing else-if-used false positive"""

if 1 == 1:
    print("FOO" + ("BAR" if 2 == 2 else ""))  # pylint does not mark this line
    print(f'FOO{"BAR" if 2 == 2 else ""}')  # pylint marks this line as else-if-used
elif 1 == 0:
    print("NOTPRINTED")
if 1 == 1:
    print("THIRD")
```
3. Run pylint against it.

### Current behavior
`test.py:6:5: R5501: Consider using "elif" instead of "else if" (else-if-used)`

### Expected behavior
No `else-if-used` error should show since `else-if` is not used anywhere.

### pylint --version output
Latest pylint/astroid preview release:
```
pylint 2.1.1
astroid 2.1.0-dev
Python 3.6.6 (default, Sep 12 2018, 18:26:19) 
[GCC 8.0.1 20180414 (experimental) [trunk revision 259383]]
```
With pylint 1.9.3, the error message is similar:
`R:  6, 5: Consider using "elif" instead of "else if" (else-if-used)`
```
pylint 1.9.3, 
astroid 1.6.5
Python 3.6.6 (default, Sep 12 2018, 18:26:19) 
[GCC 8.0.1 20180414 (experimental) [trunk revision 259383]]
```

**Comment by PCManticore** [comment]
Thanks, I can reproduce this bug.

**Comment by scottschreckengaust** [comment]
maybe also using re.match with r-strings too?

**Comment by PCManticore** [comment]
This was fixed sometime ago via https://github.com/PyCQA/pylint/commit/095928cceaef476f2e76a64a63cf215ffa782f5c.
